src/settings/settings.go

```diff
@@ -11,38 +11,45 @@ type Settings struct {
+	StatsdPort                          int    `envconfig:"STATSD_PORT" default:"8125"`
+	RuntimePath                         string `envconfig:"RUNTIME_ROOT" default:"/srv/runtime_data/current"`
+	RuntimeSubdirectory                 string `envconfig:"RUNTIME_SUBDIRECTORY"`
+	RuntimeIgnoreDotFiles               bool   `envconfig:"RUNTIME_IGNOREDOTFILES" default:"false"`
+	RuntimeWatchRoot                    bool   `envconfig:"RUNTIME_WATCH_ROOT" default:"true"`
+	LogLevel                            string `envconfig:"LOG_LEVEL" default:"WARN"`
+	LogFormat                           string `envconfig:"LOG_FORMAT" default:"text"`
+	RedisSocketType                     string `envconfig:"REDIS_SOCKET_TYPE" default:"unix"`
+	RedisType                           string `envconfig:"REDIS_TYPE" default:"SINGLE"`
+	RedisUrl                            string `envconfig:"REDIS_URL" default:"/var/run/nutcracker/ratelimit.sock"`
+	RedisPoolSize                       int    `envconfig:"REDIS_POOL_SIZE" default:"10"`
+	RedisAuth                           string `envconfig:"REDIS_AUTH" default:""`
+	RedisTls                            bool   `envconfig:"REDIS_TLS" default:"false"`
+	RedisPipelineWindow                 time.Duration
+	RedisPipelineWindowInInt64          int64  `envconfig:"REDIS_PIPELINE_WINDOW" default:"0"`
+	RedisPipelineLimit                  int    `envconfig:"REDIS_PIPELINE_LIMIT" default:"0"`
+	RedisPerSecond                      bool   `envconfig:"REDIS_PERSECOND" default:"false"`
+	RedisPerSecondSocketType            string `envconfig:"REDIS_PERSECOND_SOCKET_TYPE" default:"unix"`
+	RedisPerSecondType                  string `envconfig:"REDIS_PERSECOND_TYPE" default:"SINGLE"`
+	RedisPerSecondUrl                   string `envconfig:"REDIS_PERSECOND_URL" default:"/var/run/nutcracker/ratelimitpersecond.sock"`
+	RedisPerSecondPoolSize              int    `envconfig:"REDIS_PERSECOND_POOL_SIZE" default:"10"`
+	RedisPerSecondAuth                  string `envconfig:"REDIS_PERSECOND_AUTH" default:""`
+	RedisPerSecondTls                   bool   `envconfig:"REDIS_PERSECOND_TLS" default:"false"`
+	RedisPerSecondPipelineWindow        time.Duration
+	RedisPerSecondPipelineWindowInInt64 int64   `envconfig:"REDIS_PERSECOND_PIPELINE_WINDOW" default:"0"`
+	RedisPerSecondPipelineLimit         int     `envconfig:"REDIS_PERSECOND_PIPELINE_LIMIT" default:"0"`
+	ExpirationJitterMaxSeconds          int64   `envconfig:"EXPIRATION_JITTER_MAX_SECONDS" default:"300"`
+	LocalCacheSizeInBytes               int     `envconfig:"LOCAL_CACHE_SIZE_IN_BYTES" default:"0"`
+	NearLimitRatio                      float32 `envconfig:"NEAR_LIMIT_RATIO" default:"0.8"`
+}
+
+func (s *Settings) ConvertRedisPipelineValueToTimeDuration() {
+	s.RedisPipelineWindow = time.Duration(s.RedisPipelineWindowInInt64) * time.Second
+	s.RedisPerSecondPipelineWindow = time.Duration(s.RedisPerSecondPipelineWindowInInt64) * time.Second
 }
```

Comment:
Instead of a new config field, can you just try to convert from each type? I think that would be OK? Also I'm not sure how this couldn't wouldn't overwrite the other settings if set.

Reply:
I did another changes, by removing this function and replacing the type for both `RedisPipelineWindow` and `RedisPerSecondPipelineWindow` to be `int64` instead of `time.Duration`. Then do the conversion in the `NewRateLimiterCacheImplFromSettings` in the `redis/cache_impl.go`

Kindly let me know if you are good with this approach.